client/state/selectors/is-site-checklist-complete.js

```diff
@@ -0,0 +1,54 @@
+/** @format */
+
+/**
+ * Internal dependencies
+ */
+import getSiteChecklist from 'state/selectors/get-site-checklist';
+import isSiteChecklistLoading from 'state/selectors/is-site-checklist-loading';
+import { getSiteFrontPage } from 'state/sites/selectors';
+
+/**
+ * Checks whether the tasklist has been completed.
+ *
+ * @param  {Object}  state  Global state tree
+ * @param  {Number}  siteId Site ID
+ * @return {String} null if there's no data yet, false if the tasklist is incomplete, true if it's complete.
+ */
+export default function isSiteChecklistComplete( state, siteId ) {
+	const siteChecklist = getSiteChecklist( state, siteId );
+
+	if (
+		isSiteChecklistLoading( state, siteId ) ||
+		null === siteChecklist ||
+		! Array.isArray( siteChecklist.tasks )
+	) {
+		return null;
+	}
+
+	// True if a static page has been set as the front page, false otherwise.
+	const hasFrontPageSet = !! getSiteFrontPage( state );
+
+	/**
+		If a task is ready, it's because:
+		A) the task is marked as complete, its isCompleted prop is true.
+		B) the task front_page_updated is pending but the site doesn't have a page set as front page.
+		This is because updating the front page doesn't apply when the site doesn't have a page set as the front page. 
+		Any other case leads to a pending task.
+
+		@param   {Object}  task The task that we'll check to see if it's completed.
+		@returns {Boolean}      Whether the task is considered to be completed or not.
+	*/
+	const isTaskReady = task => {
```

Comment:
I don't think this matters too much as there's a great explanation, but is there a better word than `Ready`? I'm struggling to think of one. `Fulfilled`, `Concluded`, `Closed`, `Finished` - nothing seems quite right!

Reply:
Completed. Not sure why I wrote ready 😅